### PR TITLE
Update descriptions about Gatsby as an SSG to better describe as both an SSG + PWA

### PR DESCRIPTION
**Pull request by dominicfallows**
Creating this PR to start a conversation about the descriptor for Gatsby. After using Gatsby as my primary tech for well over a year now, I've noticed that I still have to keep having conversations about the fact that Gatsby is so much more than a static site generator. It's a full-on PWA generator, and that, IMO, shouldn't be understated.

I've been following the great work of the core Gatsby team and notice that they have been having [workshops about Gatsby and the company](https://twitter.com/dominicfallows/status/1034024103307632640) so I understand that this PR and convo might be already taking place, or that I missed a similar conversation elsewhere. 

In reality, we _could_ generalise and describe any site made through Gatsby as a PWA (thanks to the optimised way Gatsby loads static, then dynamic components), but there is probably a period of time before that term means as much to the wider audience as "static site" so I've proposed that the descriptor evolves to include SSG and PWA keywords. 

**Gatsby, a blazing-fast static site and PWA (Progressive Web App) generator for React**

A bit more 'wordy' indeed, and definitely needs group thought, hence starting a conversation.

**Gatsby, a blazing fast web app generator** might be more simple, but again does it say enough?

### CODE REVIEW

File: docs/tutorial/part-one/index.md [hunk 1]
```diff
@@ -343,7 +343,7 @@ The Gatsby `<Link />` component is for linking between pages within your site. F
 
 ## Deploying a Gatsby site
 
-Gatsby.js is a _static site generator_, which means there are no servers to setup or complicated databases to deploy. Instead, the Gatsby `build` command produces a directory of static HTML and JavaScript files which you can deploy to a static site hosting service.
+Gatsby.js is a _static site and PWA (Progressive Web App) generator_, which means there are no servers to setup or complicated databases to deploy. Instead, the Gatsby `build` command produces a directory of static HTML and JavaScript files which you can deploy to a static site hosting service.
```
Comment (wuweiweiwu) [inline docs/tutorial/part-one/index.md]: I don't think this is necessarily true, a static site can be a PWA, a PWA is not necessarily a static site. And why is this distinction important? Seems to bloat the tag line

Comment (dominicfallows) [reply docs/tutorial/part-one/index.md]: Thanks for your thoughts, it's not necessarily about describing the outputs (static site or PWA) - hence starting a convo and admitting the tag-line would need work - I absolutely agree with you that its too bloated.

I think the important point for me is that we evolve and grow how we describe Gatsby. Describing it as a simply a static site generator does it a massive disservice. 

**A blazing fast web app generator** might be more simple, but again does it say enough?

File: README.md [hunk 2]
```diff
@@ -11,7 +11,7 @@
   ⚛️ 📄 :rocket:
 </h3>
 <p align="center">
-  <strong>Blazing fast site generator for React</strong><br>
+  <strong>Blazing-fast static site and PWA (Progressive Web App) generator for React</strong><br>
```
[the remark this answers is not in the document]
Comment (dominicfallows) [reply README.md]: Thanks for the link, I was following the grammar format used elsewhere, this was a place it was different so I thought I'd update. Happy to change either way.

[the remark this answers is not in the document]
Comment (m-allanson) [reply README.md]: Let's go with the no-hyphen version: _Blazing fast_

File: docs/tutorial/part-one/index.md (hunk 1 above)
Comment (m-allanson) [reply docs/tutorial/part-one/index.md]: Thanks @dominicfallows! Getting this messaging right is something that's currently being wrestled with :) Check out the home page on [next.gatsbyjs.org](https://next.gatsbyjs.org/) for another iteration of the message. 

Could you update your PR to use the following tagline instead?

> Blazing fast modern site generator for React.js

Hopefully that's concise enough to cover what Gatsby is, without cornering it as a plain SSG.

File: packages/gatsby/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "gatsby",
-  "description": "React.js Static Site Generator",
+  "description": "React.js Static Site and PWA (Progressive Web App) Generator",
```
Comment (m-allanson) [inline packages/gatsby/package.json]: This one has the old message on it

**Comment by dominicfallows** [comment]
@m-allanson updated

**Comment by m-allanson** [comment]
Thanks @dominicfallows 👍

**Pull request merged**